apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableReadRowsRequestManager.cs

```diff
@@ -94,34 +94,44 @@ private RowSet FilterRows()
-            {
-                RowKeys = { originalRows.RowKeys.Where(key => !StartKeyIsAlreadyRead(key)) }
-            };
+            RowSet newRowSet;
 
-            foreach (RowRange rowRange in originalRows.RowRanges)
+            if (originalRows != null)
             {
-                RowRange.EndKeyOneofCase endKeyOneofCase = rowRange.EndKeyCase;
-                if (endKeyOneofCase == RowRange.EndKeyOneofCase.EndKeyClosed &&
-                    EndKeyIsAlreadyRead(rowRange.EndKeyClosed)
-                    || endKeyOneofCase == RowRange.EndKeyOneofCase.EndKeyOpen &&
-                    EndKeyIsAlreadyRead(rowRange.EndKeyOpen))
+                newRowSet = new RowSet
                 {
-                    continue;
-                }
+                    RowKeys = {originalRows.RowKeys.Where(key => !StartKeyIsAlreadyRead(key))}
+                };
 
-                RowRange.StartKeyOneofCase startKeyOneofCase = rowRange.StartKeyCase;
-                RowRange newRange = rowRange;
-                if (startKeyOneofCase == RowRange.StartKeyOneofCase.StartKeyClosed &&
-                    StartKeyIsAlreadyRead(rowRange.StartKeyClosed)
-                    || (startKeyOneofCase == RowRange.StartKeyOneofCase.StartKeyOpen &&
-                        StartKeyIsAlreadyRead(rowRange.StartKeyOpen))
-                    || startKeyOneofCase == RowRange.StartKeyOneofCase.None)
+                foreach (RowRange rowRange in originalRows.RowRanges)
                 {
-                    newRange = newRange.Clone();
-                    newRange.StartKeyOpen = LastFoundKey.Value;
+                    RowRange.EndKeyOneofCase endKeyOneofCase = rowRange.EndKeyCase;
+                    if (endKeyOneofCase == RowRange.EndKeyOneofCase.EndKeyClosed &&
+                        EndKeyIsAlreadyRead(rowRange.EndKeyClosed)
+                        || endKeyOneofCase == RowRange.EndKeyOneofCase.EndKeyOpen &&
+                        EndKeyIsAlreadyRead(rowRange.EndKeyOpen))
+                    {
+                        continue;
+                    }
+
+                    RowRange.StartKeyOneofCase startKeyOneofCase = rowRange.StartKeyCase;
+                    RowRange newRange = rowRange;
+                    if (startKeyOneofCase == RowRange.StartKeyOneofCase.StartKeyClosed &&
+                        StartKeyIsAlreadyRead(rowRange.StartKeyClosed)
+                        || (startKeyOneofCase == RowRange.StartKeyOneofCase.StartKeyOpen &&
+                            StartKeyIsAlreadyRead(rowRange.StartKeyOpen))
+                        || startKeyOneofCase == RowRange.StartKeyOneofCase.None)
+                    {
+                        newRange = newRange.Clone();
+                        newRange.StartKeyOpen = LastFoundKey.Value;
+                    }
+
+                    newRowSet.RowRanges.Add(newRange);
                 }
-                newRowSet.RowRanges.Add(newRange);
+            }
+            else
```

Comment:
Could we rewrite this as:

```csharp
// If the original request didn't specify a row set at all, create an open range from the last key
// to specify "all rows after this key".
if (originalRows == null)
{
    return RowSet.FromRowRanges(RowRange.Open(LastFoundKey, null));
}

// Rest of code as before
```

That "early out" reduces the level of nesting for the rest of the method - and probably reduces this PR to only a tiny change.

Reply:
Done. Thank you.